From 2c01ea964f53aba93afaf22e3e63b2ec30e2e3a0 Mon Sep 17 00:00:00 2001
From: Marcin Gurtowski <halftough29a@gmail.com>
Date: Mon, 2 Aug 2021 16:54:37 +0000
Subject: [PATCH] Fix creating thumbnails for items under desktop:/

By using KFileItem::localPath() which will return url.toLocalFile()
or UDS_LOCAL_PATH, so it works for all URLS.

BUG: 438691
FIXED-IN: 5.85
---
 src/widgets/previewjob.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widgets/previewjob.cpp b/src/widgets/previewjob.cpp
index f8584f735..9db3f908b 100644
--- a/src/widgets/previewjob.cpp
+++ b/src/widgets/previewjob.cpp
@@ -591,7 +591,7 @@ void PreviewJob::slotResult(KJob *job)
             id = statJob->statResult().numberValue(KIO::UDSEntry::UDS_DEVICE_ID, 0);
         }
         d->deviceIdMap[path] = id;
-        d->createThumbnail(d->currentItem.item.url().toLocalFile());
+        d->createThumbnail(d->currentItem.item.localPath());
         return;
     }
     case PreviewJobPrivate::STATE_GETORIG: {
-- 
GitLab

